### PR TITLE
fix(link-replacer): add common helper to replace link references

### PR DESCRIPTION
**Pull request by Arinono**
## Authors
@Arinono 

## Issues
refs withthegrid/platform-client#1368

## Implementation details

Mostly taking what was on both the client and the platform, refactoring it a bit and adding unit tests.

Screenshot 2022-07-15 at 19 46 06

### CODE REVIEW

File: src/commons/link-replacer.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const issue = ['issue'] as const;
+const report = ['report', 'rapport', 'meetrapport'] as const;
```
[the remark this answers is not in the document]
Comment (Arinono) [reply src/commons/link-replacer.ts]: 609f12a7817899cd5a1512d66a2bb1562fb77576

File: src/commons/link-replacer.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  ],
+  [
+    'reports',
+    new RegExp(`(?<key>${report.join('|')}) #(?<hashid>[\\w\\d]{6,})`, 'gi'),
+  ],
+  [
+    'commands',
+    new RegExp(`(?<key>${command.join('|')}) #(?<hashid>[\\w\\d]{6,})`, 'gi'),
+  ],
+];
```
[the remark this answers is not in the document]
Comment (Arinono) [reply src/commons/link-replacer.ts:19]: EDIT: Here is an example of the "failing" regexp we used before this change

Spacing in the replacement is borked
Screenshot 2022-07-18 at 11 01 16

**Review by srieding — APPROVED** [review]
Looks clean and polished

File: package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "build:open-api-file": "ts-node -P tsconfig.cjs.json --files ./documentation/",
     "build:redoc-index-html": "redoc-cli bundle ./documentation/redoc.json --output ./docs/index.html -t ./documentation/html-template.hbs",
     "prepare": "npm run build",
-    "lint": "npm run build:cjs && eslint --ext .ts src",
+    "lint": "npm run build:cjs && eslint --ext .ts src test",
```
[the remark this answers is not in the document]
Comment (Arinono) [reply package.json:30]: Yes, I wanted the test files to also be formatted/linted

**Pull request merged**